src/app/main.rs

```diff
@@ -116,13 +118,41 @@ where
             outbound_listener,
             control_listener,
             admin_listener,
+            make_refine: dns::DefaultRefine,
```

Comment:
If you instantiate the resolver in `new`, you can crate the refine with it, rather than making it a parameter on the trait.

Reply:
This doesn't really work for the reasons I described in https://github.com/linkerd/linkerd2-proxy/pull/234#discussion_r276359677 --- `dns::Resolver::from_system_config_with` returns a background future. Unfortunately, we cannot construct the background future in `new` and store it as a struct field to be spawned later, since it's an `impl Future`, a constraint imposed on us by `trust-dns`.

In fairness, the background future being an `impl Future` _is_ [my fault](https://github.com/bluejekyll/trust-dns/commit/84e6f1df30eda892663553949837f6c7dd788f69), so I suspect Ben would accept a PR changing it to a named type, but I wasn't sure if it was worth it.

Reply:
It would be fine to NewType(Box) it for now to unblock yourself